syntax/parser.go

```diff
@@ -386,7 +417,11 @@ func (p *parser) countCaptures() error {
 					}
 				} else {
 					if !p.useOptionN() && !p.ignoreNextParen {
-						p.noteCaptureSlot(p.consumeAutocap(), pos)
+						if p.useMaintainCaptureOrder() {
+							p.noteCaptureName(fmt.Sprint(p.autocap), pos)
```

Comment:
why not `consumeAutocap()` ?

Reply:
Because `consumeAutocap` is called inside `noteCaptureName` when `MaintainCaptureOrder` is enabled.

Does this need a comment somewhere?